contrib/tools/config-docs-generator/generate-config-docs.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+# Configuration - Allow environment variable overrides
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="${PROJECT_ROOT:-$(cd "$SCRIPT_DIR/../../../" && pwd)}"
+OUTPUT_DIR="$PROJECT_ROOT/docs/generated"
+CARGO_TARGET_DIR="${CARGO_TARGET_DIR:-$PROJECT_ROOT/target}"
+TEMP_DIR="${TEMP_DIR:-$CARGO_TARGET_DIR/doc-generation}"
+
+# Binary paths - allow override via environment
+EXTRACT_DOCS_BIN="${EXTRACT_DOCS_BIN:-$CARGO_TARGET_DIR/release/extract-docs}"
+GENERATE_MARKDOWN_BIN="${GENERATE_MARKDOWN_BIN:-$CARGO_TARGET_DIR/release/generate-markdown}"
+
+# Template and mappings paths - allow override via environment
+TEMPLATE_PATH="${TEMPLATE_PATH:-$SCRIPT_DIR/templates/reference_template.md}"
+SECTION_MAPPINGS_PATH="${SECTION_MAPPINGS_PATH:-$SCRIPT_DIR/section_name_mappings.json}"
+
+# Check if binaries are pre-built (skip build step)
+SKIP_BUILD="${SKIP_BUILD:-false}"
+
+export CARGO_TARGET_DIR
+
+log_info() {
+    echo -e "${GREEN}[INFO]${NC} $1"
+}
+
+log_warn() {
+    echo -e "${YELLOW}[WARN]${NC} $1"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+cleanup() {
+    if [[ -d "$TEMP_DIR" ]]; then
+        rm -rf "$TEMP_DIR"
+    fi
+}
+
+trap cleanup EXIT
+
+main() {
+    log_info "Starting config documentation generation..."
+
+    # Create necessary directories
+    mkdir -p "$OUTPUT_DIR"
+    mkdir -p "$TEMP_DIR"
+
+    cd "$PROJECT_ROOT"
+
+    # Step 1: Build the documentation generation tools
+    if [[ "$SKIP_BUILD" != "true" ]]; then
+        log_info "Building documentation generation tools..."
+        cargo build --package config-docs-generator --release
+    fi
+
+    # Step 2: Extract documentation from source code using rustdoc
+    log_info "Extracting configuration documentation using rustdoc..."
+    EXTRACTED_JSON="$TEMP_DIR/extracted-config-docs.json"
+    # List of specific Rust struct names to be documented
+    # NOTE: This variable must be manually updated if this list changes
+    # (e.g., new config structs are added or removed from the project)
+    TARGET_STRUCTS="BurnchainConfig,NodeConfig,MinerConfig,ConnectionOptionsFile,FeeEstimationConfigFile,EventObserverConfigFile,InitialBalanceFile"
```

Comment:
Is there any chance these values may change in the future? 
could we autogenerate this at runtime is the real question i'm asking .